File: tools/read_architect.lua
local luapython = require("luapython")

return {
    name = "read_architect",
    description = "Read ARCHITECT.md.",
    tool = {
        type = "function",
        ["function"] = {
            name = "read_architect",
            description = "Read ARCHITECT.md",
            parameters = {
                type = "object",
                properties = {},
                required = luapython.list {}
            }
        }
    },
    action = function(input)
        local file_name = "ARCHITECT.md"
        -- Read content from the specified file
        local file, errmsg = io.open(file_name, "r")
        if file then
            local content = file:read("*a")
            file:close()
            return { success = true, content = content }
        else
            return { success = true, content = "File is empty. " }
        end
    end
}

File: tools/read_architect.py
"""read_architect tool — read ARCHITECT.md."""

name = "read_architect"
description = "Read ARCHITECT.md design document"
schema = {
    "type": "function",
    "function": {
        "name": "read_architect",
        "description": "Read ARCHITECT.md design document",
        "parameters": {"type": "object", "properties": {}, "required": []},
    },
}


def action(input: dict) -> dict:
    try:
        with open("ARCHITECT.md", "r") as f:
            content = f.read()
        return {"success": True, "content": content}
    except FileNotFoundError:
        return {"success": True, "content": "File is empty."}
    except Exception as e:
        return {"success": False, "content": f"Failed to read: {e}"}
